### PR TITLE
fix: issues found on QA

### PR DESCRIPTION
**Pull request by tuliomir**
This PR fixes the minor bugs found during QA
### Acceptance Criteria
- A failed validation is trying to hide the modal before it's shown ( [6dca6c5](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/6dca6c5b44a93a24182caa5147e9fb2c1eaf690e) )
- "Add token" modals not being able to correctly read the token balance ( [8569b78](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/8569b78620e0ded7fa735f674a8c1b32139c107d) )
- "Change server" failures not raising messages on some cases ( [8bed03d](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/8bed03d81b2aba2d99c79e1f899dce936649b84e) )
- Unit test adjustments after merging the Atomic Swap interface PR ( [9ceeb24](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/9ceeb24ddd96ab979cbed56ce738d2dcc57196a2) )
- Fix cypress timeouts during CI ( [faeb71d](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/faeb71db8cfdc685410d934330eef5098004c12e) )
- Improves navigation to and from `permission` screen ( [9f0f71c](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/9f0f71cc7c37dfce9838759d4a76245fa2b81d1e) )

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/screens/Server.js [hunk 1]
```diff
@@ -186,6 +186,7 @@ class Server extends React.Component {
         this.executeServerChange();
       }
     } catch (e) {
+      this.state.errorMessage = e.message;
```
Comment (pedroferreira1) [inline src/screens/Server.js:189]: This is not how we set the state, we use `this.setState`. The same for the other line

File: src/screens/SentryPermission.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ class SentryPermission extends React.Component {
     if (this.state.savedPermission === null) {
       this.props.history.push('/wallet_type/');
     } else {
-      this.props.history.push('/settings/');
+      this.props.history.goBack();
```
Comment (pedroferreira1) [inline src/screens/SentryPermission.js:48]: Why did you change this? Is it a bug?

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/screens/Server.js (hunk 1 above)
Comment (tuliomir) [reply src/screens/Server.js]: My bad! Fixed on [85fd49b](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/85fd49ba3a87ed079ab314a99a95bf85470e2aeb)

File: src/screens/Server.js [hunk 3]
```diff
@@ -227,7 +228,8 @@ class Server extends React.Component {
     const promise = wallet.changeServer(this.props.wallet, pin, this.props.history);
     promise.then(() => {
       this.props.history.push('/wallet/');
-    }, () => {
+    }, (err) => {
+      this.state.errorMessage = err.message;
       this.setState({ loading: false });
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/screens/Server.js]: Done on [85fd49b](https://github.com/HathorNetwork/hathor-wallet/pull/395/commits/85fd49ba3a87ed079ab314a99a95bf85470e2aeb)

**Review by andreabadesso — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]
